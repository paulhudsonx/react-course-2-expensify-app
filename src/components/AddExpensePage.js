import React from 'react';

import {connect} from 'react-redux';
import ExpenseForm from "./ExpenseFm";
import {addExpense} from "../actions/expenses";

export class AddExpensePage extends React.Component {

  onSubmit = (expense) => {
    console.log(expense)
    this.props.addExpense(expense)
    this.props.history.push("/")
  }

  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit}/>
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => ({
  // ... computed data from state and optionally ownProps
})

/*const mapDispatchToProps = (dispatch) => {
  return {
    onSubmit: (expense) => {
      dispatch(addExpense(expense))
    }
  }
}*/

export default connect(mapStateToProps, {addExpense})(AddExpensePage)

