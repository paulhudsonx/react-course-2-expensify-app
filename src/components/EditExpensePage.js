import React from "react";

import { connect } from "react-redux";
import ExpenseForm from "./ExpenseFm";
import { startRemoveExpense, startUpdateExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    console.log(expense);
    this.props.startUpdateExpense({ id: this.props.expense.id }, expense);
    this.props.history.push("/");
    console.log(`onSubmit called!`);
  };

  onRemove = (event) => {
    console.log(`Removing ${this.props.expense.id}`);
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        Editing the expense with id of {this.props.expense.id}
        <h1>Edit Expense</h1>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  console.log(`Mapping ${props.match.params.id}`);
  return {
    expense: state.expenses.find((e) => {
      console.log(`Finding ${e.id}`);
      return e.id === props.match.params.id;
    }),
  };
};

export default connect(mapStateToProps, {
  startUpdateExpense,
  startRemoveExpense,
})(EditExpensePage);
