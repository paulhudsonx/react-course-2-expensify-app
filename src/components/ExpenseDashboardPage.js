import React from "react";
import ConnectedExpenseList from "./ExpenseList";
import ConnectedExpenseListFilters from "./ExpenseListFilters";
import ExpenseSummary from "./ExpensesSummary";

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseSummary />
    <ConnectedExpenseListFilters />
    <ConnectedExpenseList />
  </div>
);

export default ExpenseDashboardPage;
