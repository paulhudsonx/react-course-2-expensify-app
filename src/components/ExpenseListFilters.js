import React from "react";

import {connect} from 'react-redux';
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from "../actions/filters";
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };

  onDatesChange = ({startDate, endDate}) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChanged = (calendarFocused) => {
    this.setState(() => ({calendarFocused}));
  };

  onTextChange = (event) => {
    console.log(event.target.value);
    this.props.setTextFilter(event.target.value);
  }

  onSortChange = (event => {
    const sortBy = event.target.value;
    switch (sortBy) {
      case 'date':
        this.props.sortByDate();
        break;
      case 'amount':
        this.props.sortByAmount();
        break;
      default:
        this.props.sortByDate();
        break;
    }
  })

  render() {
    return (
      <div>
        <input
          type='text'
          value={this.props.filters.text}
          onChange={this.onTextChange}/>
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
          endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
          onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
          focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
          onFocusChange={this.onFocusChanged}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps, {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate})(ExpenseListFilters);
