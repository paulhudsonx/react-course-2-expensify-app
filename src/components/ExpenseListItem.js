import React from "react";

import { Link } from "react-router-dom";
import moment from "moment";
import { currencyFormat } from "../i18n/format";

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h3 className="list-item__title">{description}</h3>
      <span className="list-item__subtitle">
        {moment(createdAt).format(`MMMM Do, YYYY`)}
      </span>
    </div>
    <h3 className="list-item__data">{currencyFormat(amount / 100)}</h3>
  </Link>
);

export default ExpenseListItem;
