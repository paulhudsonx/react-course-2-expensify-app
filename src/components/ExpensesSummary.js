import React from "react";

import {connect} from 'react-redux';

import expensesTotal from "../selectors/expenses-total";
import selectExpenses from "../selectors/expenses";
import {currencyFormat} from "../i18n/format";

export const ExpensesSummary = ({expenseCount, expensesTotal}) => (
  <div>
    {expenseCount === 0 && <h1>No matching expenses!</h1>}
    {expenseCount === 1 && <h1>Viewing {expenseCount} expense totalling {currencyFormat(expensesTotal / 100)}</h1>}
    {expenseCount > 1 && <h1>Viewing {expenseCount} expenses totalling {currencyFormat(expensesTotal / 100)}</h1>}
  </div>
)

const mapStateToProps = (state) => {

  const expenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: expenses.length,
    expensesTotal: expensesTotal(expenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary)

