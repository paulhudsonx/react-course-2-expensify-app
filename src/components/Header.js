import React from "react";
import { connect } from "react-redux";
import { NavLink } from "react-router-dom";
import { startLogout } from "../actions/auth";

export const Header = (props) => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/dashboard" activeClassName="is-active" exact={true}>
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active">
      Create Expense
    </NavLink>
    <NavLink to="/edit" activeClassName="is-active">
      Edit Expense
    </NavLink>
    <NavLink to="/help" activeClassName="is-active">
      Help
    </NavLink>
    <button onClick={props.startLogout}>Logout</button>
  </header>
);

export default connect(undefined, { startLogout })(Header);
