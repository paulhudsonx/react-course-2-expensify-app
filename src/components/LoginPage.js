import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export class LoginPage extends React.Component {
  constructor(props) {
    super(props);
  }

  onSubmit = () => {};
  render = () => (
    <div>
      <h1>Login to Expensify</h1>

      <button onClick={this.props.startLogin}>Login</button>
    </div>
  );
}

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, { startLogin })(LoginPage);
