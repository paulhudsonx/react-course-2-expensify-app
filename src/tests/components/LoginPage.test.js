import React from "react";
import { shallow } from "enzyme";

//import ReactShallowRenderer from 'react-test-renderer/shallow';
import { LoginPage } from "../../components/LoginPage";

test("Should render Header correctly", () => {
  const wrapper = shallow(<LoginPage />);

  expect(wrapper).toMatchSnapshot();
});

it("should call startLogin on button click", () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin()} />);
  wrapper.find("button").simulate("click");
  expect(startLogin).toHaveBeenCalledTimes(1);
});
