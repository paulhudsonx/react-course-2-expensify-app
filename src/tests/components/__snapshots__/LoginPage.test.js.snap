// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<div>
  <h1>
    Login to Expensify
  </h1>
  <button>
    Login
  </button>
</div>
`;
